### PR TITLE
[IMP] mail: check composer state is not altered by other topbar buttons

### PR DESCRIPTION
**Pull request by xdu-odoo**
https://www.odoo.com/web#id=2243327&action=3206&active_id=1519&model=project.task&view_type=form&cids=1&menu_id=4720
task-2243327

### CODE REVIEW

File: addons/mail/static/src/messaging/component/chatter_topbar/chatter_topbar_tests.js [hunk 1]
```diff
@@ -394,6 +394,73 @@ QUnit.test('attachment counter with attachments', async function (assert) {
...
 });
 
+QUnit.test('composer state conserved when clicking on another topbar button', async function (assert) {
+    assert.expect(8);
+
+    await this.start({
+        async mockRPC(route) {
+            if (route.includes('ir.attachment/search_read')) {
+                return [];
+            }
```
Comment (alexkuhn) [inline addons/mail/static/src/messaging/component/chatter_topbar/chatter_topbar_tests.js]: can't rely on `mock_server:_mockSearchReadController`?

File: addons/mail/static/src/messaging/component/chatter_topbar/chatter_topbar_tests.js [hunk 2]
```diff
@@ -394,6 +394,73 @@ QUnit.test('attachment counter with attachments', async function (assert) {
...
+    assert.containsOnce(
+        document.body,
+        `.o_ChatterTopbar_buttonAttachments`,
+        "should have an attachments button in chatter menu"
+    );
+    await afterNextRender(() => {
+        document.querySelector(`.o_ChatterTopbar_buttonLogNote`).click();
+    });
+
+    assert.containsOnce(
```
Comment (alexkuhn) [inline addons/mail/static/src/messaging/component/chatter_topbar/chatter_topbar_tests.js]: empty line before simulated user interaction, no empty line before following assertion (same comment below)

File: addons/mail/static/src/messaging/component/chatter_topbar/chatter_topbar_tests.js (hunk 1 above)
Comment (xdu-odoo) [reply addons/mail/static/src/messaging/component/chatter_topbar/chatter_topbar_tests.js]: yes indeed, I adapt other tests of the file

**Comment by xdu-odoo** [comment]
@alexkuhn anything else ?

**Review by alexkuhn — APPROVED** [review]

**Pull request merged**